ods_ci/utils/scripts/fetch_tests.py

```diff
@@ -0,0 +1,164 @@
+import argparse
+import os
+
+from robot.model import SuiteVisitor
+from robot.running import TestSuiteBuilder
+from util import execute_command
+
+
+class TestCasesFinder(SuiteVisitor):
+    def __init__(self):
+        self.tests = []
+
+    def visit_test(self, test):
+        self.tests.append(test)
+
+
+def get_repository(test_repo):
+    """
+    If $test_repo is a remote repo, the function clones it in "ods-ci-temp" directory.
+    If $test_repo is a local path, the function checks the path exists.
+    """
+    repo_local_path = "./ods_ci/ods-ci-temp"
+    if "http" in test_repo or "git@" in test_repo:
+        print("Cloning repo ", test_repo)
+        ret = execute_command(
+            "git clone {test_repo} {output_dir}".format(test_repo=test_repo, output_dir=repo_local_path)
+        )
+        if "error" in ret.lower():
+            # actual error gets printed during "execute_command"
+            raise Exception("Failed to clone the given repository")
+    elif not os.path.exists(test_repo):
+        raise FileNotFoundError("local path {} was not found".format(test_repo))
+    else:
+        print("Using local repo ", test_repo)
+        repo_local_path = test_repo
+    return repo_local_path
+
+
+def checkout_repository(ref):
+    """
+    Checkouts the repository at current directory to the given branch/commit ($ref)
+    """
+    ret = execute_command("git checkout {}".format(ref))
+    if "error" in ret.lower():
+        # actual error gets printed during "execute_command"
+        raise Exception("Failed to checkout to the given branch/commit {}".format(ref))
+    ret = execute_command("git checkout")
+    print(ret)
+
+
+def extract_test_cases_from_ref(repo_local_path, ref):
+    """
+    Navigate to the $test_repo directory, checkouts the target branch/commit ($ref) and extracts
+    the test case titles leveraging RobotFramework TestSuiteBuilder() and TestCasesFinder() classes
+    """
+    curr_dir = os.getcwd()
+    try:
+        os.chdir(repo_local_path)
+        checkout_repository(ref)
+        builder = TestSuiteBuilder()
+        testsuite = builder.build("ods_ci/tests/")
+        finder = TestCasesFinder()
+        tests = []
+        testsuite.visit(finder)
+        for test in finder.tests:
+            # print (f'"{test.tags}"') # for future reference in order to fetch test tags
+            tests.append(test.name)
+    except Exception as err:
+        print(err)
+        os.chdir(curr_dir)
+        raise
+    os.chdir(curr_dir)
+    return tests
+
+
+def generate_rf_argument_file(tests, output_filepath):
+    """
+    Writes the RobotFramework argument file containing the test selection args
+    to include the extracted new tests in previous stage of this script.
+    """
+    content = ""
+    for testname in tests:
+        content += '--test "{}"\n'.format(testname.strip())
```

Comment:
noticed that test names may contain quotes

```
--test "Verify That "Stop Idle Notebook" Setting Is Not Overwritten After Restart Of Operator Pod"
```

we need to escape this, probably using `shlex`

```
import shlex
test_name = 'Verify That "Stop Idle Notebook" Setting Is Not Overwritten After Restart Of Operator Pod'

>>> print(f'--test {shlex.quote(test_name.strip())}')
--test 'Verify That "Stop Idle Notebook" Setting Is Not Overwritten After Restart Of Operator Pod'
```

Reply:
we actually don't need any quoting. Just tried

Reply:
fix pushed. Thanks

Reply:
Interesting. Maybe robot argfile works differently. When pasting this into commandline, I'm quite sure quoting is important. See this

```
$ echo Verify That "Stop Idle Notebook" Setting Is Not Overwritten After Restart Of Operator Pod
Verify That Stop Idle Notebook Setting Is Not Overwritten After Restart Of Operator Pod
```

quotes have disappeared!

Reply:
yeah, but I've executed using robot command and it doesn't lose quotes